### PR TITLE
Add support for AWS CloudFormation templates

### LINKED ISSUE #109: Create JUnit5 extension for deploying AWS CloudFormation templates
Per JBoss EAP testing strategy in cloud we need an option to deploy AWS CloudFormation template.

Requirements:

    The template file is located in resources
    Parameter values are settable
    JUni5 technology is used
    Extension manages lifecycle of deployed resources - delete them after tests are done

### PR DESCRIPTION
**Pull request by istraka**
Support for AWS CloudFormation teplates

Minor changes:
- Azure ARM template annotation JavaDoc enhancement
- Some checks for Azure properties added.

Fixes #109

### CODE REVIEW

File: junit5/src/main/java/sunstone/core/SunstoneCloudDeploy.java [hunk 1]
```diff
@@ -39,6 +40,23 @@ private static String md5sum(String str) throws NoSuchAlgorithmException, Unsupp
...
+    static void handleAwsCloudFormationAnnotations(ExtensionContext ctx) throws Exception {
+        SunstoneStore store = StoreWrapper(ctx);
+        AwsCloudFormationCloudDeploymentManager deploymentManager = new AwsCloudFormationCloudDeploymentManager(store.getAwsCfClient());
+        store.setAwsCfDemploymentManager(deploymentManager);
+        AutoCloseable closeable = () -> {
+            deploymentManager.undeployAll();
+            deploymentManager.close();
+        };
+        WithAwsCfTemplate[] annotations = ctx.getRequiredTestClass().getAnnotationsByType(WithAwsCfTemplate.class);
+        for (int i = 0; i < annotations.length; i++) {
```
Comment (mnovak1) [inline junit5/src/main/java/sunstone/core/SunstoneCloudDeploy.java:52]: Is it possible to process annotations in parallel?

**Review by mnovak1 — APPROVED** [review]
LGTM. Just one question.

**Review by LittleJohnII — APPROVED** [review]
LGTM.

**Comment by istraka** [comment]
Merging...

**Pull request merged**